### DISCUSSION TITLE
Multiple File Uploads (i.e. list[UploadFile]) errors, uses `body` instead of form fields

### DISCUSSION THREAD

**Discussion by sarchila** [fastapi/fastapi#11771] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import FastAPI, File, UploadFile

app = FastAPI()


#### Individual files
@app.post("/file/")
async def create_file(file: Annotated[bytes, File()]):
    return {"file_size": len(file)}


@app.post("/uploadfile/")
async def create_upload_file(file: UploadFile):
    return {"filename": file.filename}


##### Multiple files
@app.post("/files/")
async def create_files(files: Annotated[list[bytes], File()]):
    return {"file_sizes": [len(file) for file in files]}


@app.post("/uploadfiles/")
async def create_upload_files(files: list[UploadFile]):
    return {"filenames": [file.filename for file in files]}
```

### Description

- Run server with FastAPI app listed in example code (which was taken from [docs](https://fastapi.tiangolo.com/tutorial/request-files/#multiple-file-uploads) page), for my example I ran locally on port 8000.
- Try uploading test files against the different endpoints:
  - Individual file bytes: `curl -F 'file=@test.pdf' http://localhost:8000/file/`
    - returns 200: `{"file_size":202725}`
  - Individual upload file: `curl -F 'file=@test.pdf' http://localhost:8000/uploadfile/`
    - returns 200: `{"filename":"test.pdf"}`
  - Multiple upload bytes: `curl -F 'files[]=@test.pdf' http://localhost:8000/files/`
    - returns 422: `{"detail":[{"type":"missing","loc":["body","files"],"msg":"Field required","input":null}]}`
  - Multiple upload file: `curl -F 'files[]=@test.pdf' http://localhost:8000/uploadfiles/`
    - returns 422: `{"detail":[{"type":"missing","loc":["body","files"],"msg":"Field required","input":null}]}`

### Operating System

macOS

### Operating System Details

MacOS 13.5

### FastAPI Version

0.111.0

### Pydantic Version

2.7.0

### Python Version

3.11.2

**Comment by AIdjis** [comment]
for your question i updated your code it working very well just for two endpoints for you example you do not need more then 2 endpoints

```python
from fastapi import FastAPI,Query,status,HTTPException,Depends,File,UploadFile,Form
from typing import Annotated
app = FastAPI()

# for Individual files
@app.post("/uploadfile",status_code=status.HTTP_200_OK)
async def create_upload_file(file: UploadFile=File(...)):
    return {"filename": file.filename,"filesize":len(file.file.read())}
# for multiple files
@app.post("/uploadfiles",status_code=status.HTTP_200_OK)
async def create_upload_files(files: list[UploadFile]=File(...)):
    return {"filename": [file.filename for file in files],"filesize":[len(file.file.read()) for file in files]}
``` 
i hope i answered you question.

**Comment by sarchila** [comment]
Thanks for looking.  I only added the four different versions for demonstration purposes to show that the bytes and UploadFile versions, defined exactly as they are in the docs, both work for the individual case but are both not working for the multiple files case.

Your suggested changes for the `/uploadfiles` POST endpoint yield the same 422 status with the same response for me:
```json
{"detail":[{"type":"missing","loc":["body","files"],"msg":"Field required","input":null}]}
```

**Reply by AIdjis** [reply]
No, it is working very well. Just copy my code; do not change yours. The documentation you have might be outdated, but what I provided is the latest method for uploading files. Also, try using Swagger UI—it provides a better way to send your data. Here is the response body for multiples files:

```json

{
  "filename": [
    "sunset-1373171_1280.jpg"
  ],
  "filesize": [
    93161
  ]
}

```

**Reply by sarchila** [reply]
I had copied your code, so now I'm wondering what versions of FastAPI, Pydantic, and Python you're on. Same as me?  Thanks

**Reply by AIdjis** [reply]
i am using fastapi latest version :0.111.0 and pydantic =2.7.1 and my python version is 3.10.12

**Comment by YuriiMotov** [comment, marked as answer]
The problem is in `curl` command. Use `files=@test.pdf` instead of `files[]=@test.pdf`:

```
curl -F 'files=@test1.pdf' -F 'files=@test2.pdf' http://localhost:8000/files/
```
